src/openassetio-python/tests/package/hostApi/test_manager.py

```diff
@@ -1916,6 +1920,619 @@ def test_when_called_with_None_data_then_TypeError_is_raised(
+
+    @pytest.mark.parametrize(
+        "error_code,expected_exception", batch_element_error_code_exception_pairs
+    )
+    def test_when_BatchElementError_then_appropriate_exception_raised(
+        self,
+        manager,
+        mock_manager_interface,
+        a_host_session,
+        a_ref,
+        a_context,
+        error_code,
+        expected_exception,
+        a_traitsdata,
+    ):
+        method = mock_manager_interface.mock.register
+
+        expected_index = 213
+        batch_element_error = BatchElementError(error_code, "some string ✨")
+
+        def call_callbacks(*_args):
+            # Success
+            callback = method.call_args[0][5]
+            callback(expected_index, batch_element_error)
+            pytest.fail("Exception should have short-circuited this")
+
+        method.side_effect = call_callbacks
+
+        with pytest.raises(expected_exception, match=batch_element_error.message) as exc:
+            manager.register(a_ref, a_traitsdata, a_context)
+
+        method.assert_called_once_with(
+            [a_ref],
+            [a_traitsdata],
+            a_context,
+            a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        assert exc.value.index == expected_index
+        assert_BatchElementError_eq(exc.value.error, batch_element_error)
+
+
+class Test_Manager_register_with_singular_throwing_overload:
+    def test_when_register_success_then_single_EntityReference_returned(
+        self,
+        manager,
+        mock_manager_interface,
+        a_host_session,
+        a_ref,
+        a_context,
+        a_different_ref,
+        a_traitsdata,
+    ):
+        method = mock_manager_interface.mock.register
+
+        def call_callbacks(*_args):
+            # Success
+            callback = method.call_args[0][4]
+            callback(0, a_different_ref)
+
+        method.side_effect = call_callbacks
+
+        actual_ref = manager.register(
+            a_ref,
+            a_traitsdata,
+            a_context,
+            Manager.BatchElementErrorPolicyTag.kException,
+        )
+
+        method.assert_called_once_with(
+            [a_ref],
+            [a_traitsdata],
+            a_context,
+            a_host_session,
+            mock.ANY,
+            mock.ANY,
+        )
+
+        assert actual_ref == a_different_ref
+
+    @pytest.mark.parametrize(
+        "error_code,expected_exception", batch_element_error_code_exception_pairs
+    )
+    def test_when_BatchElementError_then_appropriate_exception_raised(
+        self,
+        manager,
+        mock_manager_interface,
+        a_host_session,
+        a_ref,
+        a_context,
+        error_code,
+        expected_exception,
+        a_traitsdata,
+    ):
+        method = mock_manager_interface.mock.register
+
+        expected_index = 213
+        batch_element_error = BatchElementError(error_code, "some string ✨")
+
+        def call_callbacks(*_args):
+            # Success
```

Comment:
[fix] Can we remove these comments? They're frequently stale. We should def prioritize they ticket to switch to an explicit function.

Reply:
Removed for the code introduced in this PR in c1cdd42d